.buildkite/pipeline.build.yml

```diff
@@ -648,11 +627,119 @@
+        python ./ci/run/bazel_sharding/bazel_sharding.py --exclude_manual --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}" --tag_filters=ray_air,-torch_1_11,-gpu_only,-gpu,-needs_credentials,-hdfs python/ray/train/... &&
+        python ./ci/run/bazel_sharding/bazel_sharding.py --exclude_manual --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}" --tag_filters=ray_air python/ray/data/...;
+      } > test_shard.txt
+    - cat test_shard.txt
+    - bazel test --config=ci $(./ci/run/bazel_export_options)
+      $(cat test_shard.txt)
+
+
+# These tests install requirements_py37_compat.txt which are the
+# upper bound of Python 3.7 dependencies we support
+- label: ":cold_face: :python: Ray Python 3.7 ML compatibility tests"
+  conditions:
+    ["RAY_CI_PYTHON_DEPENDENCIES_AFFECTED", "RAY_CI_TUNE_AFFECTED", "RAY_CI_TRAIN_AFFECTED", "RAY_CI_ML_AFFECTED"]
+  instance_size: large
+  parallelism: 3
+  commands:
+    - cleanup() { if [ "${BUILDKITE_PULL_REQUEST}" = "false" ]; then ./ci/build/upload_build_info.sh; fi }; trap cleanup EXIT
+    - ./ci/env/install-minimal.sh 3.7
+    - PYTHON=3.7 DATA_PROCESSING_TESTING=1 TUNE_TESTING=1 TRAIN_TESTING=1 INSTALL_HDFS=1 ./ci/env/install-dependencies.sh
+    - pip install -r python/requirements/compat/requirements_py37_compat.txt
+    - pip install -U typing-extensions
+    - HOROVOD_WITH_GLOO=1 HOROVOD_WITHOUT_MPI=1 HOROVOD_WITHOUT_MXNET=1 HOROVOD_WITH_TENSORFLOW=1 HOROVOD_WITH_PYTORCH=1 pip install horovod
+    - ./ci/env/env_info.sh
+    # Combine shards from different files
+    - >-
+      set -x;
+      {
+        python ./ci/run/bazel_sharding/bazel_sharding.py --exclude_manual --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}" --tag_filters=compat python/ray/tests/horovod/... python/ray/tests/lightgbm/... python/ray/tests/ml_py37_compat/... python/ray/tests/xgboost/... python/ray/tests/ray_lightning/... &&
         python ./ci/run/bazel_sharding/bazel_sharding.py --exclude_manual --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}" --tag_filters=-gpu,-needs_credentials,-hdfs python/ray/air/... &&
         python ./ci/run/bazel_sharding/bazel_sharding.py --exclude_manual --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}" --tag_filters=ray_air,-torch_1_11,-gpu_only,-gpu,-needs_credentials,-hdfs python/ray/train/... &&
         python ./ci/run/bazel_sharding/bazel_sharding.py --exclude_manual --index "\${BUILDKITE_PARALLEL_JOB}" --count "\${BUILDKITE_PARALLEL_JOB_COUNT}" --tag_filters=ray_air python/ray/data/...;
       } > test_shard.txt
     - cat test_shard.txt
     - bazel test --config=ci $(./ci/run/bazel_export_options)
       $(cat test_shard.txt)
+
+
+- label: ":cold_face: :python: :brain: Python 3.7 RLlib: tests/ dir"
```

Comment:
Why do we pick these tests for RLlib? They seem like a random subset of RLlib tests to me.

If we test for RLlib on 3.7, from my point of view, "RLlib: RLModule tests" and "RLlib: Examples" may be included to also test our new RLModule/Learner stack on 3.7 until it is fully deprecated. Examples also cover some extravagant stuff.

For RLlib, I would probably not fix issues that arise from these just for 3.7 if anything comes up, but it helps to be aware to better handle issues.

Reply:
I've chatted with @kouroshHakha and those were the tests he initially suggested. Generally it's ok to add more tests here, but it would be good to keep the area relatively small. RLModule + examples will add another 9 test jobs.

Of course, if this is required we should do it, but generally the idea is that "Legacy support" means we ensure basic functionality, but not necessarily compatibility with latest features.

Reply:
Got it, thanks 🙁